sklearn/linear_model/stochastic_gradient.py

```diff
@@ -284,6 +352,8 @@ def fit_binary(est, i, X, y, alpha, C, learning_rate, max_iter,
     penalty_type = est._get_penalty_type(est.penalty)
     learning_rate_type = est._get_learning_rate_type(learning_rate)
 
+    validation_mask = est._make_validation_split(X, y, sample_weight)
+
     # XXX should have random_state_!
```

Comment:
the object should store a `self.random_state_` attribute, instead of starting again from `self.random_state` each time.
I think this is unrelated to this PR.